ci-operator/step-registry/upi/install/gcp/upi-install-gcp-commands.sh

```diff
@@ -102,9 +109,14 @@ if [[ -s "${SHARED_DIR}/xpn.json" ]]; then
   HOST_PROJECT_COMPUTE_SERVICE_ACCOUNT="$(jq -r '.computeServiceAccount' "${SHARED_DIR}/xpn.json")"
   HOST_PROJECT_CONTROL_SERVICE_ACCOUNT="$(jq -r '.controlServiceAccount' "${SHARED_DIR}/xpn.json")"
   HOST_PROJECT_PRIVATE_ZONE_NAME="$(jq -r '.privateZoneName' "${SHARED_DIR}/xpn.json")"
-else
-  # Set HOST_PROJECT to the cluster project so commands with `--project` work in both scenarios.
-  HOST_PROJECT="${PROJECT_NAME}"
+  PROJECT_OPTION="--project ${HOST_PROJECT}"
+
+  if [[ -f "${CLUSTER_PROFILE_DIR}/xpn_project_setting.json" ]]; then
+    echo "Using XPN configurations from cluster profile '${CLUSTER_PROFILE}'..."
+    PROJECT_OPTION="--project ${HOST_PROJECT} --account ${GOOGLE_CLOUD_XPN_SA}"
+    BASE_DOMAIN="$(jq -r '.baseDomain' "${SHARED_DIR}/xpn.json")"
+    BASE_DOMAIN_ZONE_NAME="$(jq -r '.baseDomainZoneName' "${SHARED_DIR}/xpn.json")"
```

Comment:
The `BASE_DOMAIN` and `BASE_DOMAIN_ZONE_NAME` is the only two extra new vars compared with dev's xpn.json, right?

Reply:
Right.